include/Mathematical/SquareMatrix.hpp

```diff
@@ -311,9 +327,81 @@ class SquareMatrix : public MatrixX<_Scalar> {
 
         return value;
     }
-};
 
 
+    /**
+     *  @return an non-pivoted LU decomposition in an array, with L at position 0 and U on position 1 of the array.
+     *   warning: Pivoting is required to ensure that the decomposition is stable. Eigen3 provides partial and full pivot modules.
+     */
+    std::array<Self, 2> NoPivotLUDecomposition() const {
+
+        auto M = this->get_dim();
+
+        Self L = Self::Zero(M, M);
+        Self U = Self::Zero(M, M);
+        
+        // Algorithm from "https://www.geeksforgeeks.org/doolittle-algorithm-lu-decomposition/"
+        for (int i = 0; i < M; i++) { 
+    
+            // Upper Triangular 
+            for (int k = i; k < M; k++) { 
+    
+                // Summation of L(i, j) * U(j, k) 
+                int sum = 0; 
+                for (int j = 0; j < i; j++) 
```

Comment:
```suggestion
                for (size_t j = 0; j < i; j++) 
```

also: add opening and closing curly braces in the for loop: `{}`